http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -24,22 +24,63 @@ import org.broadinstitute.dsde.workbench.model._
 import spray.json.{RootJsonFormat, RootJsonReader, _}
 import ca.mrvisser.sealerate
 
+import scala.util.matching.Regex
+
+sealed trait ContainerRegistry extends Product with Serializable {
+  def regex: Regex
+}
+object ContainerRegistry {
+  final case object GCR extends ContainerRegistry {
+    def regex: Regex = "(.*gcr.io)\\/(.*)\\/(.*)(\\:?.*)$".r
+
+    override def toString: String = "gcr"
+  }
+  final case object DockerHub extends ContainerRegistry {
+    def regex: Regex = ".*\\/.*$".r
+    override def toString: String = "docker hub"
+  }
+
+  val allRegistries: Set[ContainerRegistry] = sealerate.values[ContainerRegistry]
+}
+
+sealed trait JupyterDockerImage extends Product with Serializable {
```

Comment:
We'll probably want to do this for RStudio as well -- maybe add a `def clusterTool: ClusterTool` method and remove `Jupyter` from the class name?

Reply:
I renamed this to `ContainerImage`. I'll leave out `clusterTool` since there's already `ClusterImage` that binds them together.